### PR TITLE
[E2E] Re-enable custom mapping null test

### PR DESCRIPTION
**Pull request by nemanjaglumac**
This test was quarantined as flaky, but I found it to be actually wrong.

This PR fixes the test and unskips it.
The stress-test workflow confirms that it's not flaking.

- 1 x 20 ✅ https://github.com/metabase/metabase/actions/runs/6943906753
- 1 x 40 ✅ https://github.com/metabase/metabase/actions/runs/6943975267

### CODE REVIEW

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js [hunk 1]
```diff
@@ -16,7 +16,6 @@ import { SAMPLE_DATABASE } from "e2e/support/cypress_sample_database";
 
 const { ORDERS, ORDERS_ID } = SAMPLE_DATABASE;
 
-// [quarantine] - intermittently failing, possibly due to a "flickering" element (re-rendering)
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/admin/datamodel/field.cy.spec.js:19]: Unrelated to this particular test.
Just a misleading comment.

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js [hunk 2]
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
...
+
       restore("withSqlite");
       cy.signInAsAdmin();
-      const dbId = 2;
+
       withDatabase(
         dbId,
-        ({ number_with_nulls: { num }, number_with_nulls_ID }) =>
-          cy.visit(
-            `/admin/datamodel/database/${dbId}/schema/${SAMPLE_DB_SCHEMA_ID}/table/${number_with_nulls_ID}/field/${num}/general`,
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/admin/datamodel/field.cy.spec.js:146]: Using `SAMPLE_DB_SCHEMA_ID` was wrong. SQLite has a different schema.

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js [hunk 3]
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
...
+      const dbId = 2;
+      const remappedNullValue = "nothin";
+
       restore("withSqlite");
       cy.signInAsAdmin();
-      const dbId = 2;
+
       withDatabase(
         dbId,
-        ({ number_with_nulls: { num }, number_with_nulls_ID }) =>
```
Comment (nemanjaglumac) [inline e2e/test/scenarios/admin/datamodel/field.cy.spec.js:144]: These values were always undefined because they exist only as all caps fields.

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js [hunk 4]
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
...
+          );
+
+          cy.log("Change `null` to custom mapping");
+          cy.findByRole("heading", { name: "Display values" })
+            .closest("section")
+            .findByText("Use original value")
+            .click();
+          popover().findByText("Custom mapping").click();
+
+          cy.findByDisplayValue("null").clear().type(remappedNullValue).blur();
```
Comment (WiNloSt) [inline e2e/test/scenarios/admin/datamodel/field.cy.spec.js:163]: I think `.blur()` is unnecessary. When you click the save button in Cypress the `blur` event should be called anyway. And mostly I found we use `blur` when we're autosaving the field, which is not what we do in this case.

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js [hunk 5]
```diff
@@ -134,39 +133,42 @@ describe("scenarios > admin > datamodel > field", () => {
...
+          cy.button("Save").click();
+          cy.button("Saved!");
```
Comment (WiNloSt) [inline e2e/test/scenarios/admin/datamodel/field.cy.spec.js:165]: We should add an assertion to avoid confusion. Using `testing-library` to find an element without acting or asserting on it is discouraged.
```suggestion
          cy.button("Saved!").should("be.visible");
```

**Review by WiNloSt — APPROVED** [review]
I have some minor comments regarding the tests, but this works great!

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js (hunk 5 above)
Comment (nemanjaglumac) [reply e2e/test/scenarios/admin/datamodel/field.cy.spec.js]: I'm actually in a group of people who prefer implicit assertions, but I hear you.
Good call.

Thank you <3

**Comment by nemanjaglumac** [comment]
> I have some minor comments regarding the tests, but this works great!

Thanks for the thorough review, as always.
Addressed the comments and ran another stress-test, just to make sure the test still passes without hiccups.
40/40 ✅ https://github.com/metabase/metabase/actions/runs/6954567753

File: e2e/test/scenarios/admin/datamodel/field.cy.spec.js (hunk 5 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/admin/datamodel/field.cy.spec.js]: ➕, I'm fine with implicit assertions

Comment (WiNloSt) [reply e2e/test/scenarios/admin/datamodel/field.cy.spec.js]: @uladzimirdev I really couldn't find where the discussion was, but explicit assertions have no disadvantages.

It could be ambiguous what an author intends to do with the element because there could be multiple things you could do with an element:
1. Assert that it's visible, not visible, not in the DOM, is in the DOM, etc.
2. Click it
3. type in it (if it's an input)
etc.

And I wouldn't know if there's no explicit action on an element whether you wanted to assert it or wanted to click it but forgot to add the `.click()` or something else.

In short, I still think it's worth being explicit with this.
